### PR TITLE
fix(provider): since Ubuntu 15.04 should use Systemd

### PR DESCRIPTION
**Pull request by fjg**
- set valid service_provider depending on Ubuntu version
- set service_name

### CODE REVIEW

**Comment by chef-supermarket** [comment]
Hi. I am an automated pull request bot named Curry. There are commits in this pull request whose authors are not yet authorized to contribute to Chef Software, Inc. projects or are using a non-GitHub verified email address. To become authorized to contribute, you will need to sign the Contributor License Agreement (CLA) as an individual or on behalf of your company. [You can read more on Chef's blog.](https://www.getchef.com/blog/2014/06/23/changes-to-the-contributor-license-agreement-process)

## GitHub Users Who Are Not Authorized To Contribute

The following GitHub users do not appear to have signed a CLA:

* @fjg

[Please sign the CLA here.](https://supermarket.chef.io/become-a-contributor)

**Comment by chef-supermarket** [comment]
Hi. Your friendly Curry bot here. Just letting you know that all commit authors have become authorized to contribute. I have added the "Signed CLA" label to this issue so it can easily be found in the future.

File: libraries/helpers.rb [hunk 1]
```diff
@@ -1,18 +1,23 @@
...
+  # helpers for the various service providers on Ubuntu systems                                                                                                                                                                               
+  module Helpers                                                                                                                                                                                                                              
+    def declare_rsyslog_service                                                                                                                                                                                                               
+      if  'ubuntu' == node['platform']                                                                                                                                                                                                        
+        if Chef::VersionConstraint.new('>= 15.04').include?(node['platform_version'])
```
Comment (tas50) [inline libraries/helpers.rb]: Is there a reason you can't use node['platform_version'] here?

Comment (fjg) [inline libraries/helpers.rb]: Not sure to understand, I am using it. See:

```
if Chef::VersionConstraint.new('>= 15.04').include?(node['platform_version']) 
```

**Pull request merged**